### PR TITLE
docs: blog post v9.0

### PR DESCRIPTION
**Pull request by xDivisionByZerox**
## Description

This PR adds a blog post page to the documentation. The page focuses on technical details that were interesting during the development of v9.

## Preview:

- https://deploy-preview-3056.fakerjs.dev/about/announcements/2024-10-26.html
- https://deploy-preview-3056.fakerjs.dev/guide/upgrading.html

### CODE REVIEW

**Comment by Shinigami92** [comment]
You need to add a link here https://github.com/faker-js/faker/blob/0fd0402aa96818c8c3af1defc816ffcffadbb1fe/docs/.vitepress/config.ts#L62-L65

File: docs/about/announcements/2024-08-31-whats-new-in-9-0.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+# What's New In v9.0
+
+::: info Looking for the migration guide?
+This article is a focusing on changes happened in the development of v9 that are on the more interesting side.
```
Comment (matthewmayer) [inline docs/about/announcements/2024-08-31-whats-new-in-9-0.md]: This article is a focusing on...

Should be

This article focuses on...

Comment (Shinigami92) [reply docs/about/announcements/2024-08-31-whats-new-in-9-0.md]: There are even more grammatical issues, so maybe just paste the blog into another spellchecking tool and fix them with one commit

File: docs/about/announcements/2024-08-31-whats-new-in-9-0.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+If you are just looking for a way to migrate from v8 to v9 check out our [migration guide](https://v9.fakerjs.dev/guide/upgrading).
+:::
+
+## Optimizing Bundle Size with Tree-Shaking
+
+In the upcoming release of Faker.js, we’ve tackled an important issue related to bundle size. The problem? Unnecessary modules were being included during tree-shaking, leading to bloated final bundles. But fear not! We’ve got a solution.
```
Comment (ST-DDT) [inline docs/about/announcements/2024-08-31-whats-new-in-9-0.md]: IMO we should not refer to upcoming release, as the blog will either be merged either right before the major release or afterwards.

File: docs/guide/upgrading.md [hunk 3]
```diff
@@ -60,50 +60,10 @@ While this is not a breaking change according to semantic versioning guidelines,
 
 ### Use High Precision RNG by Default
 
-TLDR: Many Faker methods return a different result in v9 compared to v8 for the same seed.
-
 In v9 we switch from a 32 bit random value to a 53 bit random value.
 We don't change the underlying algorithm much, but we now consume two seed values each step instead of one.
```
Comment (ST-DDT) [inline docs/guide/upgrading.md:62]: Add a link to the relevant section in the announcement blog.

File: docs/guide/upgrading.md
Comment (ST-DDT) [inline docs/guide/upgrading.md:1]: This should have a link to the announcement blog page

File: docs/about/announcements/2024-08-31-whats-new-in-9-0.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+From spices to vegetables, meats, and fruits, you can generate a wide variety of ingredients to spice up your data.
+Or make your applications more engaging with unique and mouth-watering dish names and descriptions. 
+Whether you need a single ingredient or a full menu, the Food Module offers the flexibility to meet your needs.
+
+Additionally, the [MusicModule](https://v9.fakerjs.dev/api/music.html) now featuring two brand-new methods: `album()` and `artist()`!
+These additions will enhance your music-related projects by providing more detailed and diverse data.
+Get the update and enjoy the latest tunes!
+
+## More Localized Data
+
```
Comment (ST-DDT) [inline docs/about/announcements/2024-08-31-whats-new-in-9-0.md]: This section is empty.

File: docs/about/announcements/2024-08-31-whats-new-in-9-0.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+In our continuous effort to streamline and enhance the release process for Faker, we recently implemented the first steps required for our automated release pipeline.
+This initiative aims to improve efficiency, reduce manual errors, and ensure a smoother release cycle.
+Matter of fact this version was released using our new process.
+
+Our new process consists of 4 key steps:
+
+1. [Create a release Pull Request](https://github.com/faker-js/faker/pull/2981)
+1. [Have an additional CI suite that tests against our playground](https://github.com/faker-js/faker/pull/2988)
+1. [Create a GitHub Release](https://github.com/faker-js/faker/pull/2990)
+1. [Release the latest version to npm](https://github.com/faker-js/faker/pull/2991)
```
Comment (ST-DDT) [inline docs/about/announcements/2024-08-31-whats-new-in-9-0.md]: Can we number these correctly even if GH/vitepress fixes them automatically?

File: docs/about/announcements/2024-08-31-whats-new-in-9-0.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+In our continuous effort to streamline and enhance the release process for Faker, we recently implemented the first steps required for our automated release pipeline.
+This initiative aims to improve efficiency, reduce manual errors, and ensure a smoother release cycle.
+Matter of fact this version was released using our new process.
+
+Our new process consists of 4 key steps:
+
+1. [Create a release Pull Request](https://github.com/faker-js/faker/pull/2981)
+1. [Have an additional CI suite that tests against our playground](https://github.com/faker-js/faker/pull/2988)
+1. [Create a GitHub Release](https://github.com/faker-js/faker/pull/2990)
```
Comment (ST-DDT) [inline docs/about/announcements/2024-08-31-whats-new-in-9-0.md]: IMO we should refer to this as `Draft a GitHub release`

File: docs/about/announcements/2024-08-31-whats-new-in-9-0.md (hunk 5 above)
Comment (ST-DDT) [inline docs/about/announcements/2024-08-31-whats-new-in-9-0.md]: IMO we should refer to this step as `Publish the ...`

File: docs/about/announcements/2024-08-31-whats-new-in-9-0.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+1. [Release the latest version to npm](https://github.com/faker-js/faker/pull/2991)
+
+This allows us to kick off the process with a single manual schedule of the PR creation job.
+This PR will then run an additional test CI (in parallel to our normal one) against our [playground repo](https://github.com/faker-js/playground).
+If the PR is successfully merged another job is automatically started to create a draft GitHub Release.
+This allows us to verify the correctness of the release one last time.
```
Comment (ST-DDT) [inline docs/about/announcements/2024-08-31-whats-new-in-9-0.md]: IMO this isn't the main point of that step, IMO it just gives us the ability to edit the GitHub release to our needs.
But I'm not sure whether that is nitpicking or not.

File: docs/about/announcements/2024-08-31-whats-new-in-9-0.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+1. [Have an additional CI suite that tests against our playground](https://github.com/faker-js/faker/pull/2988)
+1. [Create a GitHub Release](https://github.com/faker-js/faker/pull/2990)
+1. [Release the latest version to npm](https://github.com/faker-js/faker/pull/2991)
+
+This allows us to kick off the process with a single manual schedule of the PR creation job.
+This PR will then run an additional test CI (in parallel to our normal one) against our [playground repo](https://github.com/faker-js/playground).
+If the PR is successfully merged another job is automatically started to create a draft GitHub Release.
+This allows us to verify the correctness of the release one last time. 
+By publishing the GitHub release, the last job will kick in to publish this release to npm.
+All these steps where done manually by the maintainers, previously.
```
Comment (ST-DDT) [inline docs/about/announcements/2024-08-31-whats-new-in-9-0.md]: Maybe add a little bit of "why" here aka "and sometimes we forgot them or did them in the wrong order"

**Review by ST-DDT — COMMENTED** [review]
Nice WIP. Thanks for preparing all this.

File: docs/guide/upgrading.md
Comment (xDivisionByZerox) [reply docs/guide/upgrading.md:1]: Do you mean on the top of the file. Same as the announcement has towards the upgrading site?

**Comment by ST-DDT** [comment]
Anything else missing for this?

**Review by ST-DDT — DISMISSED** [review]

**Review by ST-DDT — DISMISSED** [review]

**Comment by ST-DDT** [comment]
(We should probably rename the file to match the current date, but otherwise I think it is good to go)

**Review by Shinigami92 — COMMENTED** [review]
I'm personally not a fan of the ` \` and also I prefer `'` over `’`, but 🤷 no blocking from my side here

I agree with the file rename, then we are potentially good to go

**Comment by ST-DDT** [comment]
> also I prefer `'` over `’`

Fixed

**Review by ST-DDT — DISMISSED** [review]

**Comment by Shinigami92** [comment]
we should rename the file before merging, because once it is merged and public, it could be referenced by blogs like dev.to or medium

**Comment by ST-DDT** [comment]
I think we should merge this and then start the release process for v9.1.

**Review by ST-DDT — APPROVED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Pull request merged**